packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1876,6 +1592,19 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 this.updateDocumentDirtyState(false);
             }
 
+            if (message.metadata?.compressed) {
+                this.logger.sendTelemetryEvent({
+                    eventName: "opToDecompress",
+                    message: JSON.stringify(message),
+                });
+
+                const contents = IsoBuffer.from(message.contents, "base64");
+                const decompressedMessage = decompress(contents);
+                const intoString = new TextDecoder().decode(decompressedMessage);
+                const asObj = JSON.parse(intoString);
+                message = { ...message, contents: asObj };
```

Comment:
This decompression is too late in processing sequence. Not sure about chunked ops (processRemoteChunkedMessage call), but decompression should happen before other parts of the system see this message, including this.pendingStateManager.processPendingLocalMessage call above